src/ui/mainwindow.cpp

```diff
@@ -1841,14 +1841,8 @@ void MainWindow::on_actionEmpty_Recent_Files_List_triggered()
 
 void MainWindow::on_actionOpen_All_Recent_Files_triggered()
 {
-    QList<QVariant> recentDocs = m_settings.General.getRecentDocuments();
-
-    QList<QUrl> convertedList;
-    for (QVariant doc : recentDocs) {
-        convertedList.append(doc.toUrl());
-    }
-
-    m_docEngine->loadDocuments(convertedList, m_topEditorContainer->currentTabWidget());
+    for (const auto& doc : m_settings.General.getRecentDocuments())
+        openRecentFileEntry(doc.toUrl());
```

Comment:
The call to `m_docEngine->loadDocuments` was useful because, in case of problems (like missing permissions for a lot of files) it offers to abort the whole operation.

Could you instead warn the user in advance if *any* of the files doesn't exist, then offer to create them anyway (all or none)? After that, you would call `m_docEngine->loadDocuments` for all the files or for only those that already existed, depending on the user choice.

I don't know if it's clear what I'm saying

Reply:
So if I understand you correctly adding a 'Yes to All' and 'No to All' option to the dialog would do what you want?

However, there's no permissions checking going on at the moment. Nqq doesn't actually create the file on the HDD until the user saves the document manually. It just opens it up as a tab and marks it as dirty. That is the same as it works right now in master. I haven't changed that part of the functionality; I added only the extra check for file existence.

Reply:
Take the master branch and follow these steps:

```bash
cd /tmp
echo a > a.txt; echo b > b.txt; echo c > c.txt
```
Now open a.txt, b.txt and c.txt in Notepadqq in order to put them in the recents list. Then close all the tabs and close Notepadqq. Now run:
```bash
chmod -r a.txt b.txt c.txt
```
and then open Notepadqq again. Choose "Open All Recent Files". You get a dialog complaining about not being able to open a.txt. You now have the option to Abort, which will abort the whole operation and will not even try to open b.txt and c.txt.

The Abort functionality is lost in this branch, because `loadDocuments` is called multiple times instead of one for all.

What I suggested was

 1. First check which files do exist, and which doesn't. Collect a list of the files which doesn't.
 2. Ask the user if he wants to open all the missing files anyway (just one dialog with a Yes/No reply).
 3. If user chose Yes, then pass the whole "recents" list to `loadDocuments`. Otherwise, pass the "recents" list minus the list of the missing files.

Reply:
I see, yes that's sensible. I implemented it that way now.